scripts/commands/checkOrphanPages.ts

```diff
@@ -0,0 +1,170 @@
+      default: false,
+      description:
+        "Check the links in the historical API docs, e.g. `api/qiskit/0.44`. " +
+        "Warning: this is slow.",
+    })
+    .parseSync();
+};
+
+async function main() {
+  const args = readArgs();
+
+  // DETERMINE THE TOC FILES TO CHECK AGAINST
+  const tocFiles = await determineTocFiles(args);
+
+  // SET PASS/FAIL STATUS
+  let allGood = true;
+  const orphanPages = [];
+
+  for (tocFile of tocFiles) {
+    // LOAD JSON CONTENTS OF TOC FILE
+    console.log("Checking toc in:", tocFile);
+    const jsonFileContents = await fs.readFile(tocFile, "utf-8");
+    const children = JSON.parse(jsonFileContents).children;
+    const files: [string] = [];
+
+    // COLLECT ALL URLS REFERENCED IN TOC FILE
+    const fileContents = await collectTocFileContents(children);
+    const flatFileContents = flattenArray(fileContents);
+    flatFileContents[0] = flatFileContents[0] + "/index";
+
+    // NOW COLLECT ALL FILES IN THE DIRECTORY
+    const dir = tocFile.replace("_toc.json", "");
+    const dirFiles = await collectDirFiles(dir);
+    for (file of dirFiles) {
+      if (!flatFileContents.includes(file)) {
+        allGood = false;
+        orphanPages.push(file);
+      }
+    }
+  }
+  if (!allGood) {
+    console.error(
+      "\n There's some orphaned pages!  These files need a home: \n",
+      orphanPages,
+    );
+    process.exit(1);
+  }
+  console.log("\nNo orphan pages found ✅\n");
+}
+
+// Collect all files in a given directory
+async function collectDirFiles(directory: string): Promise<string[]> {
+  const fileGlob = [
+    directory + "/*.md",
+    directory + "/*.mdx",
+    directory + "/*.ipynb",
+  ];
+  const fileList = await globby(fileGlob);
+  const processedFileList = fileList.map((fileName) =>
+    fileName.replace("docs", "").replace(".mdx", "").replace(".ipynb", ""),
+  );
+  return processedFileList;
+}
+
+// Collect directories to look through
+async function determineTocFiles(args: Arguments): Promise<string[]> {
+  const globs = ["docs/**/_toc.json", "!./docs/api/**"];
+  if (args.currentApis) {
+    globs.push("docs/api/*/_toc.json");
+  }
+  if (args.devApis) {
+    globs.push(
+      "docs/api/{qiskit,qiskit-ibm-runtime,qiskit-ibm-provider}/dev/_toc.json",
+    );
+  }
+  if (args.historicalApis) {
+    globs.push(
+      "docs/api/{qiskit,qiskit-ibm-provider,qiskit-ibm-runtime}/[0-9]*/_toc.json",
+    );
+  }
+  return await globby(globs);
+}
+
+// Collect files referenced in _toc file
+function collectTocFileContents(children: string[]): string[] {
+  const urls = [];
+
+  for (const child of children) {
+    // If there's any children, enter recursion to get urls
+    if ("children" in child) {
+      const childUrls = collectTocFileContents(child.children);
+      urls.push(childUrls);
+    }
+    // Just add url if there's no children
+    else {
+      urls.push(child.url);
+    }
+  }
+  return urls;
+}
+
+function flattenArray(urlArray: [string[]]): [string[]] {
```

Comment:
Nice use of `.reduce()`! Simpler is to use Lodash to do this for us: https://lodash.com/docs/#flattenDeep. You can remove this function and replace its call with Lodash.

Lodash is a great library of utilities like this.

Reply:
Thanks!  Recursion always ends up being trickier than I expect for me, but glad there's a much easier library to call from.